app/controllers/sessions_controller.rb

```diff
@@ -12,24 +10,48 @@ def destroy
     redirect_to root_url
   end
 
-  def googleAuth
-    # Manages the callback from Google
-    # Get access tokens from the google server
+  def omniauth_failure
+    redirect_to root_url
+  end
+
+  def setTokens(service, token, refresh_token, user)
+    case service
+    when :google
+      user.google_token = token
+      user.google_refresh_token = refresh_token if refresh_token.present?
+    when :microsoft
+      user.microsoft_token = token
+      user.microsoft_refresh_token = refresh_token if refresh_token.present?
+    end
```

Comment:
just curious, why distinguish the tokens? if the usage of the tokens is the same, can't we just do `user.token` and `user.refresh_token`? Since users are not supposed to change their email, this token should be bound to one service only.

Reply:
I talked to Michael about this -- he says that the tokens actually might not be necessary for how the tracker is currently being used, but if we do keep them they should be stored separately. Searching online gives me mixed answers as well so I think we should leave it like this for now.